chore(eslint-plugin): deprecate no-var-requires in favor of no-require-imports (#8334)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/no-require-imports.mdx

</TabItem>
</Tabs>

### `allowAsImport`

> When set to `true`, require statements with an `import` declaration won't be reported.
> This is useful if you're on an older CommonJS-style codebase and haven't been able to transition to ESM-style imports yet.

Reviewer: ```suggestion
When set to `true`, the `import x = require(...)` declaration won't be reported.
This is useful if you use certain module options that require strict CommonJS interop semantics.
```

</TabItem>
</Tabs>

### `allowAsImport`

If it's set to `true`, the require statements with an `import` declaration won't be reported.
This is useful if you're on an older CommonJS-style codebase and haven't been able to transition to ESM-style imports yet.

With `{allowAsImport: true}`:

<Tabs>
<TabItem value="❌ Incorrect">

> ```ts

Reviewer 2: ```suggestion
```ts option='{ "allowAsImport": true }'
```

It looks like `option` is missing here?

var foo = require('foo');
const foo = require('foo');
let foo = require('foo');
```

</TabItem>
<TabItem value="✅ Correct">

> ```ts

Reviewer 2: ```suggestion
```ts option='{ "allowAsImport": true }'
```

And here as well
